adal/src/main/java/com/microsoft/aad/adal/TokenCacheAccessor.java

```diff
@@ -286,6 +286,13 @@ void updateCachedItemWithResult(final String resource, final String clientId, fi
                 throw new AuthenticationException(ADALError.DEVELOPER_AUTHORITY_IS_NOT_VALID_URL, e.getMessage(), e);
             }
         } else if (AuthenticationConstants.OAuth2ErrorCode.INVALID_GRANT.equalsIgnoreCase(result.getErrorCode())) {
+            // make sure to set authority to the one in preferred cache
+            try {
+                mAuthority = getAuthorityUrlWithPreferredCache();
+            } catch (final MalformedURLException e) {
+                //TODO : Understand what can be done here
+                Logger.d(TAG + methodName, "Authority from preferred caches is invalid");
```

Comment:
Let's do the following logging here so that we receive the `Exception` if PII logging is enabled:

```java
com.microsoft.identity.common.internal.logging.Logger.error(TAG, "Authority from preferred caches is invalid", null)
com.microsoft.identity.common.internal.logging.Logger.errorPII(TAG, "Failed with Exception", e);
```